File: backend/models/user.model.js
const { DataTypes, INTEGER, STRING } = require("sequelize");
const sequelize = require("../config/database");
// const generateUniqueId = () => {
//   const prefix = "CDIPL"; // Your custom prefix
//   const randomSuffix = Math.floor(1000 + Math.random() * 9000); // Random number between 1000 and 9999
//   return `${prefix}${randomSuffix}`; // Combine prefix and random number
// };

const User = sequelize.define("User", {
  // id: {
  //   type: DataTypes.STRING,
  //   primaryKey: true,
  //   defaultValue: generateUniqueId,
  // },
  employeeId: {
    type: DataTypes.STRING,
    allowNull: true,
    unique: true,
  },
  phone: { type: DataTypes.BIGINT },
  name: { type: DataTypes.STRING },
  password: { type: DataTypes.STRING },
  email: { type: DataTypes.STRING },
  points: { type: DataTypes.INTEGER, defaultValue: 0 },
  // roleId: {
  //   type: DataTypes.INTEGER,
  //   defaultValue: 3,
  //   allowNull: false,
  //   references: {
  //     model: "Roles",
  //     key: "id",
  //   },
  // },

  role: {
    type: DataTypes.STRING,
    defaultValue: "user",
    enum: ["user", "admin", "superadmin"],
  },
  // meallist: { type: DataTypes.JSON, allowNull: true },
});

// User.associate = (models) => {
//   User.belongsTo(models.Role, {
//     foreignKey: "roleId",
//     as: "role",
//   });
// };

// User.sync({ force: false })
User.sync({ force: false })

  .then(() => {
    console.log("User table created");
  })
  .catch((error) => {
    console.error("Error creating User table:", error);
  });

module.exports = User;

File: cdipl/lib/models/usermodel.dart
class User {
  final int id; // Corresponds to "id" in JSON
  final String employeeId;
  final String name;
  final String email;
  final int phone;
  final String role;

  User({
    required this.id,
    required this.employeeId,
    required this.name,
    required this.email,
    required this.phone,
    required this.role,
  });

  // Factory method to create a User object from JSON
  factory User.fromJson(Map<String, dynamic> json) {
    return User(
      id: json['id'] as int,
      employeeId: json['employeeId'] as String,
      name: json['name'] as String,
      email: json['email'] as String,
      phone: json['phone'] as int,
      role: json['role'] as String,
    );
  }

  // Method to convert a User object to JSON
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'employeeId': employeeId,
      'name': name,
      'email': email,
      'phone': phone,
      'role': role,
    };
  }
}
